src/components/DayPicker.jsx

```diff
@@ -174,21 +204,107 @@ export default class DayPicker extends React.Component {
         this.adjustDayPickerHeight();
       }
     }
+
+    if (isFocused !== this.props.isFocused) {
+      if (isFocused) {
+        const focusedDate = this.getFocusedDay(currentMonth);
+        this.setState({
+          showKeyboardShortcuts,
+          focusedDate,
+        });
+      } else {
+        this.setState({ focusedDate: null });
+      }
+    }
   }
 
   shouldComponentUpdate(nextProps, nextState) {
     return shallowCompare(this, nextProps, nextState);
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (this.state.monthTransition || !this.state.currentMonth.isSame(prevState.currentMonth)) {
+    const { monthTransition, currentMonth } = this.state;
+    if (monthTransition || !currentMonth.isSame(prevState.currentMonth)) {
       if (this.isHorizontal()) {
         this.adjustDayPickerHeight();
       }
     }
+
+    if (!prevProps.showKeyboardShortcuts && this.props.showKeyboardShortcuts) {
+      // not sure what the right approach is here
```

Comment:
Basically, my thought was that if you type `?` inside the date input, probably focus should move to the `DayPicker` instead of the input... but that's pretty hard and also maybe a bad pattern?